tsl/src/nodes/vector_agg/function/functions.c

```diff
@@ -27,10 +27,13 @@ typedef struct
 } CountState;
 
 static void
-count_init(void *agg_state)
+count_init(void *restrict agg_states, int n)
 {
-	CountState *state = (CountState *) agg_state;
-	state->count = 0;
+	CountState *states = (CountState *) agg_states;
+	for (int i = 0; i < n; i++)
+	{
+		states[i].count = 0;
+	}
```

Comment:
Why should we have to pass `n` here? Wouldn't it be easier if `n` (the number of elements) be part of the agg_state that gets passed in?

For example:
```suggestion
static void
count_init(AggState *aggstate)
{
	CountState *states = (CountState *) aggstate->states;
	
	for (int i = 0; i < aggstate->num_states; i++)
	{
		states[i].count = 0;
	}
```

Or ideally, with allocation:
```suggestion
static AggState *
count_create(int num_states)
{
	AggState *state = palloc0(sizeof(AggState) + sizeof(CountState) * num_states);
	state->num_states = num_states;
	
	//  Below not really necessary with palloc0
	CountState *states = (CountState *) &aggstate->states[0];
	
	for (int i = 0; i < aggstate->num_states; i++)
	{
		states[i].count = 0;
	}
	
	return state;
```

Reply:
`n` is a number of separate aggregate states that are supposed to be initialized, e.g. when we've allocated new aggregate states for new 100 grouping keys and initialize them in bulk. It's not a part of a single aggregate state for a single grouping key.

Reply:
The proposed pattern can easily be updated to handle resize (see new suggestion is separate comment).